person.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'nameable'
+require_relative 'capitalize_decorator'
+require_relative 'trimmer_decorator'
+
+# Person class provides the default implementation of the correct_name method
+class Person < Nameable
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  # Getter and setter in one
+  attr_accessor :name, :age
+
+  # Getter
+  attr_reader :id
+
```

Comment:
- [ ] please note that A person has many rentals. Kindly ensure that A method for adding rentals should be present. here, the person itself should be sent as a parameter to create the new rental, achieving the 'has-many' association`
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```